### PR TITLE
Fix rel updates

### PR DESCRIPTION
**Pull request by ray6080**
Fix `SET` on rel tables. Also introduce the rel local storage. This is gonna break our test coverage as code path for insert/delete in local storage are not covered yet, which will be covered in the following PR.

### CODE REVIEW

**Review by andyfengHKU — APPROVED** [review]

File: src/include/common/enums/rel_direction.h [hunk 1]
```diff
@@ -1,21 +1,11 @@
...
-enum class RelDataDirection : uint8_t { FWD = 0, BWD = 1 };
-
-struct RelDataDirectionUtils {
-    static inline std::vector<RelDataDirection> getRelDataDirections() {
-        return std::vector<RelDataDirection>{RelDataDirection::FWD, RelDataDirection::BWD};
-    }
-
-    static std::string relDataDirectionToString(RelDataDirection direction);
-};
+enum class RelDataDirection : uint8_t { FWD = 0, BWD = 1, INVALID = 2 };
```
Comment (acquamarin) [inline src/include/common/enums/rel_direction.h:8]: When will the relDirection be INVALID? i think it should be either FWD or BWD, Invalid is a little bit confusing

File: src/storage/local_storage/local_table.cpp [hunk 2]
```diff
@@ -89,142 +55,53 @@ void LocalVectorCollection::prepareAppend() {
 }
 
 LocalNodeGroup::LocalNodeGroup(std::vector<LogicalType*> dataTypes, MemoryManager* mm) {
-    columns.resize(dataTypes.size());
+    chunks.resize(dataTypes.size());
```
Comment (acquamarin) [inline src/storage/local_storage/local_table.cpp:58]: Better to use reserve, and then call append in the for loop.
resize requires that the elements inside chunks have a valid copy constructor.

File: src/storage/local_storage/local_table.cpp [hunk 3]
```diff
@@ -89,142 +55,53 @@ void LocalVectorCollection::prepareAppend() {
...
-    if (!nodeGroups.contains(nodeGroupIdx)) {
-        nodeGroups.emplace(nodeGroupIdx, std::make_unique<LocalNodeGroup>(dataTypes, mm));
-    }
-    return nodeGroupIdx;
-}
-
-LocalTableData* LocalTable::getOrCreateLocalTableData(
-    const std::vector<std::unique_ptr<Column>>& columns, MemoryManager* mm) {
-    if (!localTableData) {
+    auto dataIdx = tableType == TableType::NODE ? 0 : direction == RelDataDirection::FWD ? 0 : 1;
```
Comment (acquamarin) [inline src/storage/local_storage/local_table.cpp:92]: I see this pattern: `RelDataDirection::FWD ? 0 : 1;` many times, can we refactor to a function?

File: src/processor/map/map_set.cpp [hunk 4]
```diff
@@ -75,27 +75,31 @@ std::unique_ptr<RelSetExecutor> PlanMapper::getRelSetExecutor(
...
+        auto columnID = common::INVALID_COLUMN_ID;
         if (property->hasPropertyID(tableID)) {
-            propertyID = property->getPropertyID(tableID);
+            auto propertyID = property->getPropertyID(tableID);
+            columnID =
+                catalog->getReadOnlyVersion()->getTableSchema(tableID)->getColumnID(propertyID);
         }
         return std::make_unique<SingleLabelRelSetExecutor>(
-            table, propertyID, srcNodePos, dstNodePos, relIDPos, propertyPos, std::move(evaluator));
+            table, columnID, srcNodePos, dstNodePos, relIDPos, propertyPos, std::move(evaluator));
```
Comment (acquamarin) [inline src/processor/map/map_set.cpp:102]: As we discussed, we should have a unified design about using columnID vs propertyID. Can you open an issue about this?

**Review by acquamarin — APPROVED** [review]

File: src/include/common/enums/rel_direction.h (hunk 1 above)
Comment (ray6080) [reply src/include/common/enums/rel_direction.h:8]: It was supposed to be a safe guard for assertions. but I don't think it's necessary, removed it now.

File: src/processor/map/map_set.cpp (hunk 4 above)
Comment (ray6080) [reply src/processor/map/map_set.cpp:102]: Done here .https://github.com/kuzudb/kuzu/issues/2430

**Pull request merged**